internal/handler/mqtthandler/mqtt_handler.go

```diff
@@ -70,23 +69,47 @@ func NewHandler(server, username, password, cafile string) (handler.Handler, err
 	return &h, nil
 }
 
-func newTLSConfig(cafile string) (*tls.Config, error) {
+func newTLSConfig(cafile, certFile, certKeyFile string) (*tls.Config, error) {
+	if cafile == "" && (certFile == "" || certKeyFile == "") {
+		log.Info("handler/mqtt: TLS config is empty")
+		return nil, nil
+	}
+
+	tlsConfig := &tls.Config{}
+
 	// Import trusted certificates from CAfile.pem.
+	if cafile != "" {
+		cacert, err := ioutil.ReadFile(cafile)
+		if err != nil {
+			log.Errorf("handler/mqtt: couldn't load cafile: %s", err)
+			return nil, err
+		}
+		certpool := x509.NewCertPool()
+		certpool.AppendCertsFromPEM(cacert)
 
-	cert, err := ioutil.ReadFile(cafile)
-	if err != nil {
-		log.Errorf("backend: couldn't load cafile: %s", err)
-		return nil, err
+		tlsConfig.RootCAs = certpool // RootCAs = certs used to verify server cert.
 	}
 
-	certpool := x509.NewCertPool()
-	certpool.AppendCertsFromPEM(cert)
+	// Import certificate and the key
+	if certFile != "" && certKeyFile != "" {
+		cert, err := ioutil.ReadFile(certFile)
+		if err != nil {
+			log.Errorf("handler/mqtt: couldn't load certFile: %s", err)
+			return nil, err
+		}
+
+		certKey, err := ioutil.ReadFile(certKeyFile)
+		if err != nil {
+			log.Errorf("handler/mqtt: couldn't load certKeyFile: %s", err)
+			return nil, err
+		}
+
+		kp, err := tls.X509KeyPair(cert, certKey)
```

Comment:
Please use https://golang.org/pkg/crypto/tls/#LoadX509KeyPair. It allows you to directly load the files by passing the paths to these files. Saves a couple of lines ;-)